File: sbe/binance/spot_3_2/generated-python/spot_sbe/FilterType.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple, Set
from io import BytesIO

class FilterType:
    """Enum values."""
    MAX_POSITION = 0
    PRICE_FILTER = 1
    TPLUS_SELL = 2
    LOT_SIZE = 3
    MAX_NUM_ORDERS = 4
    MIN_NOTIONAL = 5
    MAX_NUM_ALGO_ORDERS = 6
    EXCHANGE_MAX_NUM_ORDERS = 7
    EXCHANGE_MAX_NUM_ALGO_ORDERS = 8
    ICEBERG_PARTS = 9
    MARKET_LOT_SIZE = 10
    PERCENT_PRICE = 11
    MAX_NUM_ICEBERG_ORDERS = 12
    EXCHANGE_MAX_NUM_ICEBERG_ORDERS = 13
    TRAILING_DELTA = 14
    PERCENT_PRICE_BY_SIDE = 15
    NOTIONAL = 16
    MAX_NUM_ORDER_LISTS = 17
    EXCHANGE_MAX_NUM_ORDER_LISTS = 18
    MAX_NUM_ORDER_AMENDS = 19
    MAX_ASSET = 20
    NON_REPRESENTABLE = 254
    NULL_VALUE = 255

    @staticmethod
    def encoded_length() -> int:
        return 1

File: sbe/binance/spot_3_1/generated-php/spot_sbe/FilterType.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class FilterType
{
    public const MAXPOSITION = 0;
    public const PRICEFILTER = 1;
    public const TPLUSSELL = 2;
    public const LOTSIZE = 3;
    public const MAXNUMORDERS = 4;
    public const MINNOTIONAL = 5;
    public const MAXNUMALGOORDERS = 6;
    public const EXCHANGEMAXNUMORDERS = 7;
    public const EXCHANGEMAXNUMALGOORDERS = 8;
    public const ICEBERGPARTS = 9;
    public const MARKETLOTSIZE = 10;
    public const PERCENTPRICE = 11;
    public const MAXNUMICEBERGORDERS = 12;
    public const EXCHANGEMAXNUMICEBERGORDERS = 13;
    public const TRAILINGDELTA = 14;
    public const PERCENTPRICEBYSIDE = 15;
    public const NOTIONAL = 16;
    public const MAXNUMORDERLISTS = 17;
    public const EXCHANGEMAXNUMORDERLISTS = 18;
    public const MAXNUMORDERAMENDS = 19;
    public const MAXASSET = 20;
    public const NONREPRESENTABLE = 254;
    public const NULL_VALUE = 255;

    public static function encodedLength(): int
    {
        return 1;
    }
}
